extensions/data-plane/data-plane-http/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSinkFactoryTest.java

```diff
@@ -15,134 +15,148 @@
     }
 
     @Test
-    void verifyCreateAuthenticatingSource() throws InterruptedException, ExecutionException, IOException {
-        var authKey = FAKER.lorem().word();
-        var authCode = FAKER.internet().uuid();
-        var dataAddress = HttpDataAddress.Builder.newInstance()
-                .baseUrl("http://example.com")
-                .authKey(authKey)
-                .authCode(authCode)
+    void verifySuccessSinkCreation() {
+        var address = HttpDataAddress.Builder.newInstance().build();
+        var request = createRequest(address);
+        var params = HttpRequestParams.Builder.newInstance()
+                .baseUrl("http://" + FAKER.internet().url())
+                .method(HttpMethod.POST.name())
+                .contentType("application/json")
                 .build();
 
-        var validRequest = createRequest(HttpDataAddress.DATA_TYPE).destinationDataAddress(dataAddress).build();
+        when(supplierMock.apply(request)).thenReturn(params);
 
-        var call = mock(Call.class);
-        when(call.execute()).thenReturn(createHttpResponse().build());
+        assertThat(factory.validate(request).succeeded()).isTrue();
+        var sink = factory.createSink(request);
+        assertThat(sink).isNotNull();
 
-        when(httpClient.newCall(isA(Request.class))).thenAnswer(r -> {
-            assertThat(((Request) r.getArgument(0)).headers(authKey).get(0)).isEqualTo(authCode);  // verify auth header set
-            return call;
-        });
-
-        var sink = factory.createSink(validRequest);
-
-        var result = sink.transfer(new InputStreamDataSource("test", new ByteArrayInputStream("test".getBytes()))).get();
-
-        assertThat(result.failed()).isFalse();
+        var expected = HttpDataSink.Builder.newInstance()
+                .params(params)
+                .httpClient(HTTP_CLIENT)
+                .monitor(MONITOR)
+                .requestId(request.getId())
+                .executorService(EXECUTOR_SERVICE)
+                .build();
 
-        verify(call).execute();
+        // validate the generated data sink field by field using reflection
+        Arrays.stream(HttpDataSink.class.getDeclaredFields()).forEach(f -> {
+            f.setAccessible(true);
+            try {
+                assertThat(f.get(sink)).isEqualTo(f.get(expected));
+            } catch (IllegalAccessException e) {
+                throw new AssertionError("Comparison failed for field: " + f.getName());
+            }
+        });
     }
 
-    @Test
-    void verifyCreateAdditionalHeaders() throws InterruptedException, ExecutionException, IOException {
-        var dataAddress = HttpDataAddress.Builder.newInstance()
-                .baseUrl("http://example.com")
-                .contentType("application/test-octet-stream")
-                .addAdditionalHeader("x-ms-blob-type", "BlockBlob")
+    @ParameterizedTest
+    @CsvSource({ "POST", "POST", "PUT" })
```

Comment:
you can simply use @ ValueSource(strings = {"POST", "PUT"}) instead - also there are duplicate tests for "POST"